src/helpers/probabilities.js

```diff
@@ -1,157 +1,136 @@
+  let Y_mc = [];
+  let Y_ab = [];
+  let Y_bc = [];
+
+  try {
+    const locationData = await Location.findByPk(location);
+
+    if (!locationData) throw new Error("Location not found");
+
+    const latitude = locationData.latitude;
+    const longitude = locationData.longitude;
+
+    coordinates = await getDBCoordinates(1);
+
+    polygons = await getDBPolygons(1);
+
+    const polygon = getPolygon(latitude, longitude, polygons, coordinates);
+
+    if (polygon === -1) throw new Error("Polygon not found");
+
+    const ponderationsData = await W_one.findAll();
+
+    if (ponderationsData < 0) throw new Error("Ponderations not found");
+
+    let ponderations = [];
+
+    for (const pond of ponderationsData) {
+      ponderations.push([pond.Y_y, pond.Y_z, pond.Y_mc, pond.Y_ab, pond.Y_bc]);
+    }
+
+    coordinates = await getDBCoordinates(4);
+
+    polygons = await getDBPolygons(4);
+
+    const polygon2 = getPolygon(
+      latitude,
+      longitude,
+      polygons,
+      coordinates
+    );
+
+    if (polygon2 === -1) throw new Error("Polygon not found");
+
+    switch (polygon2) {
+      case 0:
+        zer_data = await Zer0.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 1:
+        zer_data = await Zer1.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 2:
+        zer_data = await Zer2.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 3:
+        zer_data = await Zer3.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 4:
+        zer_data = await Zer4.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 5:
+        zer_data = await Zer5.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 6:
+        zer_data = await Zer6.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      case 7:
+        zer_data = await Zer7.findAll({
+          where: { id: location, Periodo: period },
+        });
+        break;
+      default:
+        throw new Error("Polygon is out of parameters");
+    }
+
+    if (!zer_data) throw new Error("Information not found");
+
+    for (const data of zer_data) {
+      X.push(data.X);
+      Y_y.push(data.Y_y);
+      Y_z.push(data.Y_z);
+      Y_mc.push(data.Y_mc);
+      Y_ab.push(data.Y_ab);
+      Y_bc.push(data.Y_bc);
+    }
+
+
+    for (let i = 0; i < 20; i++) {
+      sum =
+        ponderations[polygon][0] * Y_y[i] +
+        ponderations[polygon][1] * Y_z[i] +
+        ponderations[polygon][2] * Y_mc[i] +
+        ponderations[polygon][3] * Y_ab[i] +
+        ponderations[polygon][4] * Y_bc[i];
+      sum = sum.toFixed(8);
+      result.push({ x: X[i] / 981, y: sum });
+    }
+
+    return result;
+
+  } catch (error) {
+    return error;
```

Comment:
Try / catch is needed?

Reply:
yes, it is needed to catch any errors and handle them. if not, the execution of the server is terminated on the first error.